pkg/cmd/describe/describe_rollout.go

```diff
@@ -0,0 +1,504 @@
+		return err
+	}
+
+	if !o.Watch {
+		return r.Visit(o.describeRollout)
+	}
+
+	return o.watchRollout(r)
+}
+
+func (o *DescribeRolloutOptions) describeRollout(info *resource.Info, err error) error {
+	if err != nil {
+		return err
+	}
+
+	rollout, err := o.RolloutViewerFn(info.Object)
+	if err != nil {
+		return err
+	}
+
+	o.printRolloutInfo(rollout)
+	return nil
+}
+
+func (o *DescribeRolloutOptions) watchRollout(r *resource.Result) error {
+	infos, err := r.Infos()
+	if err != nil {
+		return err
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("watch is only supported on a single rollout")
+	}
+	info := infos[0]
+
+	watcher, err := o.RolloutsClient.Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector: "metadata.name=" + info.Name,
+	})
+
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if o.TimeoutSeconds > 0 {
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(o.TimeoutSeconds)*time.Second)
+		defer cancel()
+	}
+
+	return o.watchRolloutUpdates(ctx, watcher)
+}
+
+func (o *DescribeRolloutOptions) watchRolloutUpdates(ctx context.Context, watcher watch.Interface) error {
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return nil
+			}
+			if event.Type == watch.Added || event.Type == watch.Modified {
+				if rollout, ok := event.Object.(*rolloutsapi.Rollout); ok {
+					o.clearScreen()
+					o.printRolloutInfo(rollout)
+				}
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func (o *DescribeRolloutOptions) clearScreen() {
+	fmt.Fprint(o.Out, "\033[2J\033[H")
+}
+
+func (o *DescribeRolloutOptions) GetResources(rollout *rolloutsapi.Rollout) (*WorkloadInfo, error) {
+	resources := []string{rollout.Spec.WorkloadRef.Kind + "/" + rollout.Spec.WorkloadRef.Name}
+	r := o.Builder().
+		WithScheme(internalapi.GetScheme(), scheme.Scheme.PrioritizedVersionsAllGroups()...).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(true, resources...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
+
+	if err := r.Err(); err != nil {
+		return nil, err
+	}
+
+	obj, err := r.Object()
+	if err != nil {
+		return nil, err
+	}
+
+	workloadInfo := &WorkloadInfo{}
+	objValue := reflect.ValueOf(obj).Elem()
+	workloadInfo.Name = objValue.FieldByName("Name").String()
+	workloadInfo.Kind = objValue.Type().Name()
+
+	podTemplateSpec := objValue.FieldByName("Spec").FieldByName("Template").FieldByName("Spec")
+	containers := podTemplateSpec.FieldByName("Containers")
+	for i := 0; i < containers.Len(); i++ {
+		container := containers.Index(i)
+		workloadInfo.Images = append(workloadInfo.Images, container.FieldByName("Image").String())
+	}
+
+	// Deployment,StatefulSet,CloneSet,Advanced StatefulSet,Advanced DaemonSet
+
+	switch o := obj.(type) {
+	case *appsv1.Deployment:
+		workloadInfo.Replicas.Desired = *o.Spec.Replicas
+		workloadInfo.Replicas.Current = o.Status.AvailableReplicas
```

Comment:
what current replicas means ? is it the number actual replicas, or the number of replicas of currentRevision ? 
if it is the former(i think it is), current replicas is actually o.Status.Replicas. current replicas of all workload(except daemonset) is o.Status.Replicas